### ISSUE TITLE
STL capture crash the server with timeout

### ISSUE THREAD

**Issue by hhaim** [cisco-system-traffic-generator/trex-core#365]
The tests are not stable on all setups -- issue started in may 

```
*** RPC thread has encountred an unexpected error: 'internal error: failed to get response from remote core for more than '1000' ms'

assert: ../../src/rpc-server/trex_rpc_req_resp_server.cpp:82 virtual void TrexRpcServerReqRes::_rpc_thread_cb() Assertion '0' failed.

*** traceback follows ***

1             0x551c41 __assert_fail + 673
2             0x68e204 ./_t-rex-64() [0x68e204]
3       0x7fdbae5a327f so/x86_64/libstdc++.so.6(+0xba27f) [0x7fdbae5a327f]
4       0x7fdbaec8cdd5 /lib64/libpthread.so.0(+0x7dd5) [0x7fdbaec8cdd5]
5       0x7fdbadd0202d clone + 109


*** addr2line information follows ***

basic_string.h:6289
trex_rpc_req_resp_server.cpp:82
??:0
??:0
??:0


./t-rex-64: line 86: 13530 Aborted                 (core dumped) ./_$(basename $0) $INPUT_ARGS $EXTRA_INPUT_ARGS
Killing Scapy server... Scapy server is killed
```

```

info.setup.name.keyword: Descending | info.setup.name.keyword: Descending | Count
-- | -- | --
trex21 | trex21 | 58
trex12 | trex12 | 44
trex11 | trex11 | 40
trex19 | trex19 | 40
trex16 | trex16 | 3
trex07 | trex07 | 1
  |   |  

```

**Comment by hhaim** [comment]
need to check if     ```bool Reschedule() {``` is called from CP->Rx() in the CP() context

**Comment by bdollma** [comment]
From my check, it is not. @jsmoon mentions here https://github.com/cisco-system-traffic-generator/trex-core/pull/342#issuecomment-534926098 that CP->RX is blocking and doesn't use reschedule. However, take a look here:

https://github.com/cisco-system-traffic-generator/trex-core/blob/924a6c7c39f7907d99415a4e79ddb54e20a5cce8/src/stx/stl/trex_stl_fs.cpp#L1153

In one hand, we are waiting for reply:

```cpp
        /* hold until message was ack'ed - otherwise we might lose packets */        
        reply.wait_for_reply();  
        assert(m_rx_core->is_working());
```

In the other hand, we are using `use_resched` as `true`:
```cpp
m_ring_to_rx->SecureEnqueue((CGenNode *)msg,true);
```

This was changed 3 months, ago 
```
Hanoh Haim, 3 months ago   (February 15th, 2021 3:11pm) 

Change ZMQ SUB/PUB interface. fix #636 see more info in this case

Signed-off-by: Hanoh Haim <hhaim@cisco.com>

 aa1b905            |    Discuss / Collab…   |   

- m_ring_to_rx->SecureEnqueue((CGenNode *)msg);
+ m_ring_to_rx->SecureEnqueue((CGenNode *)msg,true);
Changes   b351894  ⟷   aa1b905   |   
```
so that explains why the issue is happening again.

So in summary, if I understand correctly, we must call SecureEnqueue with `use_resched=false` each time we are sending a message from CP to Rx.
